web/src/message_list.js

```diff
@@ -129,7 +129,11 @@ export class MessageList {
             // message. Regardless of whether the messages are new or
             // old, we want to select a message as though we just
             // entered this view.
-            this.select_id(this.first_unread_message_id(), {then_scroll: true, use_closest: true});
+            this.select_id(this.first_unread_message_id(), {
+                then_scroll: true,
+                use_closest: true,
+                mark_read: false,
+            });
```

Comment:
This is called every time we successfully fetch messages from the server and we do want to mark the first unread message as read in some of those cases.

I think the main issue here is that when we're processing messages after a reload, we already have a message that was selected and that information isn't available here, so I think that's perhaps the direction we want to think about for this fix.